src/test/java/com/github/prominence/openweathermap/api/ApiTest.java

```diff
@@ -22,15 +22,21 @@
 
 package com.github.prominence.openweathermap.api;
 
+import com.github.prominence.openweathermap.api.context.ApiConfiguration;
 import org.junit.jupiter.api.BeforeAll;
 
+import java.util.Optional;
+
 public class ApiTest {
+    public static final String OPENWEATHER_API_KEY = "OPENWEATHER_API_KEY";
+    public static final String RUN_ONE_CALL = "RUN_ONE_CALL";
```

Comment:
Why do we need that static variable?

Reply:
I have added assumptions like these to the tests where the real API is called:
```java
Assumptions.assumeTrue(System.getenv(OPENWEATHER_API_KEY) != null, "Api key is not set, skip.");
Assumptions.assumeTrue(System.getenv(RUN_ONE_CALL) != null, "Skipping one-call API calls.");
```
This way, if the environment parameter is not provided, that means we don't want to run that test (as it would have no way to succeed anyway) and it will skip the test instead of failing it. The benefit of this is for those devs who have only free API key but still prefer to see green tests in the end (to not spend time on checking which tests failed exactly when we expect some tests to fail every time).